### PR TITLE
LPS-184962 Adapt RichText to accept localized values

### PR DESCRIPTION
**Pull request by pedro-oliveira446**
[LPS-184962](https://issues.liferay.com/browse/LPS-184962)
[LPS-184961](https://issues.liferay.com/browse/LPS-184961)

### CODE REVIEW

**Comment by pedro-oliveira446** [comment]
ci:test:relevant

**Comment by pedro-oliveira446** [comment]
ci:test:forms

**Comment by marcelabc** [comment]
Just started reviewing :)

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/business/type/RichTextObjectFieldBusinessType.java [hunk 1]
```diff
@@ -67,6 +69,16 @@ public String getName() {
 		return ObjectFieldConstants.BUSINESS_TYPE_RICH_TEXT;
 	}
 
+	@Override
+	public Map<String, Object> getProperties(
+		ObjectField objectField,
+		ObjectFieldRenderingContext objectFieldRenderingContext) {
+
+		return HashMapBuilder.<String, Object>put(
+			"objectFieldLocalized", objectField.isLocalized()
```
Comment (marcelabc) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/business/type/RichTextObjectFieldBusinessType.java:78]: Hey @pedro-oliveira446,
I searched the words "objectFieldLocalized" and "localizedObjectFields" in the object's modules. I saw that the option "localizedObjectFields" is the one used for variables. Also, it is better to finish the variable with the entity's name like "xyzObjectField".
Do you agree? Could you replace it please?

**Review by marcelabc — CHANGES_REQUESTED** [review]
Hi @pedro-oliveira446 
I finished my review. Great work :clap: 
Let me know if you have questions :smile:

**Comment by pedro-oliveira446** [comment]
ci:test:sf

**Comment by pedro-oliveira446** [comment]
ci:test:object

**Comment by pedro-oliveira446** [comment]
ci:test:forms

**Review by marcelabc — APPROVED** [review]
LGTM 🎉

**Comment by pedro-oliveira446** [comment]
ci:test:sf

**Comment by pedro-oliveira446** [comment]
ci:test:relevant

**Comment by pedro-oliveira446** [comment]
ci:test:objects

**Comment by pedro-oliveira446** [comment]
ci:test:forms

**Comment by pedro-oliveira446** [comment]
ci:test:object

**Comment by marcelabc** [comment]
Hey @pedro-oliveira446, 
Can we add the label Dev Approved or we need frontend review? (@naraandrade )

**Comment by pedro-oliveira446** [comment]
Hi @marcelabc,
We still need frontend review.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/util/localizable/transform.es.js [hunk 2]
```diff
@@ -25,6 +25,17 @@ export function convertValueToJSON(value) {
 	return value;
 }
 
+export function convertStringToObject(value, localeId) {
+	if (typeof value === 'string') {
+		const object = new Object();
+		object[localeId] = value;
```
Comment (naraandrade) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/util/localizable/transform.es.js:31]: @pedro-oliveira446 Please, add a blank line between the const definiton and its usage. Thanks :)

**Review by naraandrade — CHANGES_REQUESTED** [review]

**Comment by naraandrade** [comment]
@pedro-oliveira446 Great work, just a really small change needed :)

**Comment by pedro-oliveira446** [comment]
ci:test:sf

**Comment by pedro-oliveira446** [comment]
ci:test:relevant

**Comment by pedro-oliveira446** [comment]
ci:test:object

**Review by naraandrade — APPROVED** [review]

**Comment by pedro-oliveira446** [comment]
Resent here https://github.com/brianchandotcom/liferay-portal/pull/136511

**Pull request closed without merging**